cli/src/action/registry/mod.rs

```diff
@@ -177,3 +182,161 @@ impl Action for RegistryGenerateAction {
         Ok(())
     }
 }
+
+#[cfg(feature = "registry-add-node")]
+pub struct RegistryAddAction;
+
+#[cfg(feature = "registry-add-node")]
+impl Action for RegistryAddAction {
+    fn run<'a>(&mut self, arg_matches: Option<&ArgMatches<'a>>) -> Result<(), CliError> {
+        let args = arg_matches.ok_or(CliError::RequiresArgs)?;
+
+        let url = args
+            .value_of("url")
+            .map(ToOwned::to_owned)
+            .or_else(|| std::env::var(SPLINTER_REST_API_URL_ENV).ok())
+            .unwrap_or_else(|| DEFAULT_SPLINTER_REST_API_URL.to_string());
+
+        let identity = args
+            .value_of("identity")
+            .ok_or_else(|| CliError::ActionError("Identity must be specified".into()))?
+            .to_string();
+
+        let display_name = args
+            .value_of("display_name")
+            .unwrap_or(&identity)
+            .to_string();
+
+        let mut builder = SplinterRestClientBuilder::new();
+        builder = builder.with_url(url);
+
+        #[cfg(feature = "splinter-cli-jwt")]
+        {
+            let private_key = args.value_of("private_key_file");
+            builder = builder.with_auth(create_cylinder_jwt_auth(private_key)?);
+        }
+
+        let client = builder.build()?;
+
+        let metadata: Vec<(String, String)> = if let Some(metadata) = args.values_of("metadata") {
+            metadata
+                .map(|kv| {
+                    let mut kv_iter = kv.splitn(2, '=');
```

Comment:
We've had some discussions recently about how to handle parsing like this and we settled on a slightly different pattern that's a bit more clear and explicit: https://github.com/Cargill/splinter/blob/master/libsplinter/src/rest_api/auth/identity/mod.rs#L113

Reply:
Fixed